lib/src/clojure_lsp/feature/hover.clj

```diff
@@ -116,7 +116,7 @@
   (let [db @db*
         filename (shared/uri->filename uri)
         zloc (some-> (f.file-management/force-get-document-text uri db*)
```

Comment:
I think at least for hover feature, it could makes sense to have a safe or a try-catch it because if we don't find a correct zloc, hover can still be calculated from kondo analysis right? Although there is a chance to be outdated since we can't parse the code too.. so I think it's ok to throw the exception. WDYT @mainej ?

Reply:
Another thought is that that kind of exceptions may log a lot on server log, in this case, logging just a line mentioning the parse didn't work sounds better to me too, I can see users that don't use paredit like tools checking server logs and complaining about those exceptions just because for a moment their code was in a invalid state before they fix the parens for example

Reply:
When we can't parse, we don't have outdated analysis, we have no analysis. That is, we send the invalid text to clj-kondo and, along with a diagnostic about the syntax error, it returns an empty vector of analysis. So we never find any hover information to return. The net effect was that with `safe-zloc-of-string` we were creating a successful response with no results. With `zloc-of-string` we're creating an error response. From the user's perspective, it's the same–nothing happens in the editor. The only real difference is whether the logs contain a stacktrace.

That said, I agree, stacktraces in the logs are alarming even if they don't mean anything is wrong.

So maybe the better approach is to make `zloc-of-string` private, and instead use `safe-zloc-of-string` everywhere. Anywhere that relies on an Exception being thrown to short-circuit can be refactored to check for nils instead. There aren't actually many uses of either function, so it wouldn't be hard to refactor.

WDYT @ericdallo? Would that be better?

To be honest, the main reason I wanted to remove `safe-zloc-of-string` was that it was previously dead code and it caused #1080. But after the `println` was fixed there was really nothing wrong with it. So I'd be OK if we revive it.

Reply:
> When we can't parse, we don't have outdated analysis, we have no analysis. That is, we send the invalid text to clj-kondo and, along with a diagnostic about the syntax error, it returns an empty vector of analysis. So we never find any hover information to return. The net effect was that with safe-zloc-of-string we were creating a successful response with no results. With zloc-of-string we're creating an error response. From the user's perspective, it's the same—nothing happens in the editor. The only real difference is whether the logs contain a stacktrace.

Agreed

> WDYT @ericdallo? Would that be better?

Yeah, I'd like just to avoid the unnecessary logs, so sounds good to me
I agree it was dead code, but now, it may be a good opportunity to start using it in the correct way avoiding the verbose exception logs I think

Reply:
👍 I moved the switch to safe-zloc-of-string to #1092.